ng-dev/release/notes/release-notes.ts

```diff
@@ -63,9 +63,14 @@ export class ReleaseNotes {
 
   /**
    * Prepend generated release note to the CHANGELOG.md file in the base directory of the repository
-   * provided by the GitClient.
+   * provided by the GitClient. Removes entries for related prerelease entries as appropriate.
    */
   async prependEntryToChangelogFile() {
+    // When the version for the entry is a non-prelease (i.e. 1.0.0 rather than 1.0.0-next.1), the
+    // pre-release entries for the version should be removed from the changelog.
+    if (semver.prerelease(this.version) === null) {
+      Changelog.removePrereleaseEntriesForVersion(this.git, this.version);
+    }
```

Comment:
I'm not sure what it would look like if we'd manually invoke the remove pre-release function in the release action when staging (keeping this function more concerned with just prepending), but not sure 🤷‍♂️.

Anyway, have you considered adding a test that validates the removal as part of a release action? that is more reflecting what we'll end up with the release tool.

Reply:
I initially looked at doing it that way, it its a somewhat sizable refactor to bring the logic required out of the base class to allow it to be different depending on what action is running.

Since we are always wanting the first stable release notes to remove the prerelease notes, this logic of removing all prerelease notes for non-prerelease versions has the same effect.

I will add a validation test as suggested.